CONTRIBUTING.md

```diff
@@ -0,0 +1,88 @@
+# Contributing guidelines
+
+- [Introduction](#introduction)
+- [Pull Requests (PRs)](#pull-requests-prs)
+  - [Etiquette for creating PRs](#etiquette-for-creating-prs)
+  - [Checklist for publishing PRs](#checklist-for-publishing-prs)
+- [Setting up a local development
+  environment](#setting-up-a-local-development-environment)
+  - [Code checks with pre-commit](#code-checks-with-pre-commit)
+  - [Unit tests with pytest](#unit-tests-with-pytest)
+
+## Introduction
+
+Thank you for contributing to SHAP. SHAP is an open source collective effort,
+and contributions of all forms are welcome!
+
+You can contribute by:
+
+- Submitting bug reports and features requests on the GitHub issue tracker
+- Contributing fixes and improvements via Pull Requests
+
+## Pull Requests (PRs)
+
+### Etiquette for creating PRs
+
+- Before starting on a PR, please make a proposal by opening an Issue, and check
+  for any duplicates
```

Comment:
The current phrasing seems to suggest that an issue is always needed.

I think the advice here only applies for more "complex" PRs. E.g. for PRs that just fix a typo, we don't need them to raise an issue beforehand.